extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -300,4 +367,32 @@ export class AmpAd3PImpl extends AMP.BaseElement {
     }
     this.lifecycleReporter.sendPing(eventName);
   }
+
+  /**
+  * Calculates and attempts to set the appropriate height & width for a
+  * responsive full width ad unit.
+  * @return {!Promise}
+  * @private
+  */
+  attemptFullWidthSizeChange_() {
+    const viewportSize = this.getViewport().getSize();
+    const maxHeight = Math.min(MAX_FULL_WIDTH_HEIGHT, viewportSize.height);
+    const ratio = this.config.fullWidthHeightRatio;
+    dev().assert(!!ratio,
```

Comment:
this is already checked in the `shouldRequestFullWidth_` function

Reply:
Yes. 
That's why it's a dev assert and noted as "should never happen" - it's here as help for future developers.

Reply:
I don't see this duplicated check needed. again, all error message takes real bytes in the binary.